cs/cs/Serializer/VowpalWabbitDefaultMarshallerExt.tt

```diff
@@ -155,6 +155,90 @@ namespace VW.Serializer
 			context.AppendStringExample(feature.Dictify, featureString);
         }
 
+		/// <summary>
+        /// Transfers feature data to native space.
+        /// </summary>
+		/// <param name="context">The marshalling context.</param>
+		/// <param name="ns">The namespace description.</param>
+		/// <param name="feature">The feature description.</param>
+		/// <param name="value">The feature value.</param>
+        public void MarshalFeature(VowpalWabbitMarshalContext context, Namespace ns, Feature feature, IEnumerable<<#=t#>> value)
+        {
+            if (value == null)
+            {
+                return;
+            }
+
+            ulong i = 0;
+
+            // support anchor feature
+            if (feature.AddAnchor)
+            {
+                context.NamespaceBuilder.AddFeature(ns.NamespaceHash, 1);
+                i++;
+            }
+
+            foreach (var v in value)
+            {
+				<# if(new[]{ typeof(Int64), typeof(UInt64), typeof(double) }.Contains(t)) {#>
+				#if DEBUG
+				if (v > float.MaxValue || v < float.MinValue)
+				{
+					Trace.TraceWarning("Precision lost for feature value: " + v);
+				}
+				#endif
+				<# } #>
+
+                context.NamespaceBuilder.AddFeature((ulong)(ns.NamespaceHash + i), (float)v);
+                i++;
+            }
+
+            if (context.StringExample == null)
+            {
+                return;
+            }
+
+			string featureString;
```

Comment:
I've seen inconsistent whitespace issues with this file before, is it possible to fix?